tfsdk/attribute.go

```diff
@@ -344,4 +344,13 @@ func (a Attribute) validate(ctx context.Context, req ValidateAttributeRequest, r
 			return
 		}
 	}
+
+	if a.DeprecationMessage != "" {
```

Comment:
Wouldn't we only want to return the error if the value wasn't `null`?

Reply:
Yes, we would! Good catch. Testing updated to catch this:

```
--- FAIL: TestAttributeValidate (0.00s)
    --- FAIL: TestAttributeValidate/deprecation-message-null (0.00s)
        attribute_test.go:1313: Unexpected response (+wanted, -got):   tfsdk.ValidateAttributeResponse{
            -   Diagnostics: []*tfprotov6.Diagnostic{
            -           &{
            -                   Severity:  s"WARNING",
            -                   Summary:   "Attribute Deprecated",
            -                   Detail:    "Use something else instead.",
            -                   Attribute: s`AttributeName("test")`,
            -           },
            -   },
            +   Diagnostics: nil,
              }
```

Fix will be included in next commit.